artifactory/system.go

```diff
@@ -33,6 +40,1112 @@ func (v Versions) String() string {
+	WebServerType            *string `yaml:"webServerType,omitempty" xml:"webServerType,omitempty"`
+	ArtifactoryAppContext    *string `yaml:"artifactoryAppContext,omitempty" xml:"artifactoryAppContext,omitempty"`
+	PublicAppContext         *string `yaml:"publicAppContext,omitempty" xml:"publicAppContext,omitempty"`
+	ServerName               *string `yaml:"serverName,omitempty" xml:"serverName,omitempty"`
+	ServerNameExpression     *string `yaml:"serverNameExpression,omitempty" xml:"serverNameExpression,omitempty"`
+	SslCertificate           *string `yaml:"sslCertificate,omitempty" xml:"sslCertificate,omitempty"`
+	SslKey                   *string `yaml:"sslKey,omitempty" xml:"sslKey,omitempty"`
+	DockerReverseProxyMethod *string `yaml:"dockerReverseProxyMethod,omitempty" xml:"dockerReverseProxyMethod,omitempty"`
+	UseHttps                 *bool   `yaml:"useHttps,omitempty" xml:"useHttps,omitempty"`
+	UseHttp                  *bool   `yaml:"useHttp,omitempty" xml:"useHttp,omitempty"`
+	SslPort                  *int    `yaml:"sslPort,omitempty" xml:"sslPort,omitempty"`
+	HttpPort                 *int    `yaml:"httpPort,omitempty" xml:"httpPort,omitempty"`
+	ArtifactoryServerName    *string `yaml:"artifactoryServerName,omitempty" xml:"artifactoryServerName,omitempty"`
+	UpStreamName             *string `yaml:"upStreamName,omitempty" xml:"upStreamName,omitempty"`
+	ArtifactoryPort          *int    `yaml:"artifactoryPort,omitempty" xml:"artifactoryPort,omitempty"`
+}
+
+// ReverseProxies is an alias for a slice of ReverseProxy that can be
+// properly marshaled to/from YAML.
+type ReverseProxies []*ReverseProxy
+
+// MarshalYAML implements the yaml.Marshaller interface for ReverseProxies.
+func (r ReverseProxies) MarshalYAML() (interface{}, error) {
+	reverseProxiesMap := make(map[string]*ReverseProxy)
+	for _, reverseProxy := range r {
+		if *reverseProxy == (ReverseProxy{Key: reverseProxy.Key}) {
+			reverseProxiesMap[*reverseProxy.Key] = nil
+		} else {
+			reverseProxiesMap[*reverseProxy.Key] = reverseProxy
+		}
+	}
+
+	return reverseProxiesMap, nil
+}
+
+// UnmarshalYAML implements yaml.Unmarshaler for ReverseProxies.
+func (r *ReverseProxies) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	reverseProxiesMap := make(map[string]*ReverseProxy)
+	if err := unmarshal(reverseProxiesMap); err != nil {
+		return err
+	}
+
+	var reverseProxiesSlice ReverseProxies
+	for reverseProxyKey, reverseProxy := range reverseProxiesMap {
+		if reverseProxy == nil {
+			continue
+		}
+		reverseProxy.Key = &reverseProxyKey
+		reverseProxiesSlice = append(reverseProxiesSlice, reverseProxy)
+	}
+
+	*r = reverseProxiesSlice
+	return nil
+}
+
+// PropertySet represents a Property Set in Artifactory's Global Configuration Descriptor.
+//
+// Docs: https://www.jfrog.com/confluence/display/RTF/YAML+Configuration+File#YAMLConfigurationFile-General(General,PropertySets,Proxy,Mail)
+type PropertySet struct {
+	Name       *string     `yaml:"-" xml:"name,omitempty"`
```

Comment:
I noticed we're not allowing this field to be serialized in YAML due to the `-` tag.

This peaked my curiosity so was this intentional?

Reply:
Similar to [comment](https://github.com/target/go-arty/pull/57#discussion_r436119180).